components/insight/SRC/org/openmicroscopy/shoola/env/data/views/calls/ThumbnailLoader.java

```diff
@@ -220,13 +222,24 @@ protected void buildTree() {
                 BatchCall call = new BatchCall("Loading thumbnails") {
                     @Override
                     public void doCall() throws Exception {
-                        ThumbnailStorePrx store = getThumbnailStore(pxd);
+                        ThumbnailStorePrx store = null;
                         try {
+                            store = getThumbnailStore(pxd);
                             handleBatchCall(store, pxd, userId);
+                        } catch (ApiUsageException ex) {
```

Comment:
If I understand the diff correctly, in the previous version this was protected by at least `Exception` if not `Throwable` to handle various possibilities. I defer on how this is handled elsewhere in Insight, but any remote call can call a number of exceptions beyond `ApiUsageException`.

Reply:
Yes, the safest way to prevent Insight crashes would be catch everything and show a black thumbnail in any error case. I'll change that.